lib/urlapi.c

```diff
@@ -744,6 +752,24 @@ static CURLUcode seturl(const char *url, CURLU *u, unsigned int flags)
     memcpy(hostname, hostp, len);
     hostname[len] = 0;
 
+    if((flags & CURLU_GUESS_SCHEME) && !schemep) {
+      /* legacy curl-style guess based on host name */
+      if(checkprefix("ftp.", hostname))
```

Comment:
In the wild, my (completely unscientific) gut feeling is that this will fall down into the `else` case quite often nowadays. Is it worth microoptimizing this to see if the rather expensive case insensitive comparisons are at all potentially useful? Something like
```C
if(strlen(hostname) >= 3 && (*(hostname + 3) == '.' || *(hostname + 4) == '.')) {
 ...
```

Reply:
Not a bad idea, but I would prefer if we then try that out and do that optimization in a separate commit. Possibly after doing at least some basic check that it actually makes a noticeable difference... You good with that?